### ISSUE TITLE
Hitbox Issues for Rear Armour

### ISSUE THREAD

**Issue by AdrianGin** [MechWarriorOnline/issue-tracker#209]
Issue:
Shots to destroyed mech sections from behind remove front armour instead of rear armour.

1. Move behind an enemy mech.
2. Destroy Left Arm
3. Continue shooting left arm from behind. Notice how Left Torso Front Armour takes damage 
4. Continue to destroy Left Torso from behind
5. Continue shooting left torso from behind. Notice how transferred damage is done to the front armour first.

Expected/Intended outcome:
Rear armour is removed when shot from behind.

**Comment by Tyler799** [comment]
How was this testing performed? In testing grounds or a private lobby?

If it was testing grounds, I suggest you re-test in a private lobby. Testing grounds has both been proven (and confirmed by the developers themselves) as a poor environment for testing issues. It doesn't tend to act the same as a real match or private lobby.

As well, video evidence would be highly appreciated. Preferably with a high enough bitrate and framerate to clearly see the impact location and the paper doll change.

Heck, perhaps testing on a couple 'Mechs with a layout of their hitboxes (from the hitbox localization page) as an extra would be highly appreciated.

Information is ammunition. The more evidence you bring to the table the more clearly the problem shows itself for PGI to fix.

**Comment by tauSentry** [comment]
Administrivia...  

At first glance, this looks like it might be a duplicate of either #90 or #105. Can you confirm?

If it is definitely not a duplicate, can you also upload or link to a video that captures this behavior.

**Comment by AdrianGin** [comment]
Not a duplicate.
https://www.youtube.com/watch?v=mQvqF7EhT04

**Comment by Tyler799** [comment]
This is not a bug, for whatever it's worth. This is how damage transfer works in MWO, and always has since CB. 

When destroyed, continuing damage of the legs goes to the side torsos. When arms are destroyed, the continuing damage goes to the side torsos. And when ST's and legs are destroyed, damage gets moved to the CT. Every time a transfer occurs only 40% of the damage makes it.

When you shoot a destroyed arm, 40% of the damage moves to the ST. _This is the intended mechanic_ even if it looks a little weird when you shoot an arm from the rear.

**Comment by tauSentry** [comment]
@Tyler799, I think(?) what @AdrianGin is specifically pointing out is that the transferred-damage is always being applied to the front armor.

In other words, I suppose the question boils down to...
"What exactly is _supposed_ to happen when transferred-damage is applied to a Front+Rear Armored component?"  
- **A)** Apply directly to internal structure.
- **B)** Apply first to Front Armor (if any), then to Back Armor (if any), then to internal structure.
- **C)** Apply first to Back Armor (if any), then to Front Armor (if any), then to internal structure.
- **D)** Spread damage to both Front Armor and Back Armor equally until one runs out. If both run out, then apply to internal structure.
- **E)** Some other protocol not discussed here.

I couldn't immediately find a solid reference on this.

**Comment by Tyler799** [comment]
Currently, the intended method for MWO is just what's shown in-game. Shooting a stump arm transfers the damage to the ST. Front armor is damaged first, then structure. It's always been that way since CB. 

I don't have a solid reference either. It's been years since they've discussed this sort of thing. 

At first, I thought that A should be applied. You are shooting an area where no armor exists. That damage should go to ST structure, with the normal 40% rate applying. However, for ST-CT transfer, that could get a little nasty, (especially for 'Mechs with larger STs) and I could easily see an argument against it. 

Adding a second hitbox (a rear arm hitbox) isn't practical at this point in the games development. (Every single mech would need its hitboxes redone).

And doing B or C would be kind of silly and make damage transfer even less effective than currently. 

D would feel weird and non-deterministic from the players perspective. 

So personally, I think that our current system is what's best.

**Comment by AdrianGin** [comment]
How about:

Shooting destroyed section should transfer damage to the respective adjacent section from the direction in which the damage came from.





-----Original Message-----

**Comment by Tyler799** [comment]
I don't believe that MWO has any way to detect the direction that the damage is coming from, just where it falls. I could be wrong on that, but you'd need to ask one of the main devs. 

As long as it doesn't involve adding new hitboxes, I suppose it's not impractical time-wise. But whether it's feasible technically I can't answer.

This issue should be closed and re-opened as a suggestion, or re-named as such.

**Comment by AdrianGin** [comment]
There's already detection for the rear torso components, so it could be used.

**Comment by Tyler799** [comment]
There is no such detection (directional) as far as I am aware. Rear torso shots are handled by the rear-torso hitboxes. Nothing more.
